jwst/ramp_fitting/tests/test_ramp_fit.py

```diff
@@ -681,13 +682,39 @@ def test_miri_first_last():
     # Put CR in 1st (0-based) group
     model1.groupdq[0, 1, 1, 1] = JUMP_DET
 
-    new_mod, int_model, opt_model, gls_opt_model = ramp_fit(model1, 1024 * 30000.,
-                                                            True, rnModel, gain, 'OLS',
-                                                            'optimal', 'none')
+    new_mod, int_model, opt_model, gls_opt_model = ramp_fit(
+        model1, 1024 * 30000., True, rnModel, gain, 'OLS', 'optimal', 'none')
 
     np.testing.assert_allclose(new_mod.data, 10. / 3., rtol=1E-5)
 
 
+def test_miri_no_good_pixel():
+    ''' 
+    With no good data, MIRI will remove all groups where all pixels are bad.
+    If all groups are bad, NoneType is returned for all return values from
+    ramp_fit.
+    '''
+    nints, ngroups, nrows, ncols = 1, 2, 2, 2
+    deltatime = 3.
+    model1, gdq, rnModel, pixdq, err, gain = setup_small_cube(
+        ngroups, nints, nrows, ncols, deltatime)
+
+    # Dummy non-zero data to make sure if processing occurs a non-NoneType gets
+    # returned.
+    model1.data[0, :, 0, 0] = np.array([-200., -500.], dtype=np.float32)
+    model1.data[0, :, 0, 1] = np.array([-300., -600.], dtype=np.float32)
+    model1.data[0, :, 1, 0] = np.array([-200., 900.], dtype=np.float32)
+    model1.data[0, :, 1, 1] = np.array([-600., -400.], dtype=np.float32)
+
+    # For all pixels, set gdq for 0th and final groups to DO_NOT_USE
+    model1.groupdq[:, :, :, :] = DO_NOT_USE
```

Comment:
Doesn't that set ALL groups to DO_NOT_USE ?

Reply:
Yes.